fili-core/src/main/java/com/yahoo/bard/webservice/web/responseprocessors/DruidPartialDataResponseProcessor.java

```diff
@@ -0,0 +1,261 @@
+            checkOverflow(json, query);
+
+            SimplifiedIntervalList overlap = getOverlap(json, query);
+            if (!overlap.isEmpty()) {
+                logAndGetErrorCallback(ErrorMessageFormat.DATA_AVAILABILITY_MISMATCH.format(overlap), query);
+            }
+            if (next instanceof FullResponseProcessor) {
+                next.processResponse(json, query, metadata);
+            } else {
+                next.processResponse(json.get(DruidJsonResponseContentKeys.RESPONSE.getName()), query, metadata);
+            }
+        } else if (statusCode == Status.NOT_MODIFIED.getStatusCode() && !(next instanceof FullResponseProcessor)) {
+            logAndGetErrorCallback(
+                    "Content Not Modified(304), but no etag cache response processor is available to process " +
+                            "the 304 response",
+                    query);
+        } else {
+            next.processResponse(json, query, metadata);
+        }
+    }
+
+    /**
+     * Validates JSON response object to make sure it contains all of the following information.
+     * <ul>
+     *     <li>X-Druid-Response-Context
+     *         <ol>
+     *             <li>uncoveredIntervals</li>
+     *             <li>uncoveredIntervalsOverflowed</li>
+     *         </ol>
+     *     </li>
+     *     <li>status-code</li>
+     * </ul>
+     *
+     * @param json  The JSON response that is to be validated
+     * @param query  The query with the schema for processing this response
+     */
+    private void validateJsonResponse(JsonNode json, DruidAggregationQuery<?> query) {
+        if (json.getNodeType() == JsonNodeType.ARRAY) {
+            logAndGetErrorCallback("JSON response is missing X-Druid-Response-Context and status code", query);
+        }
+
+        if (!json.has(DruidJsonResponseContentKeys.DRUID_RESPONSE_CONTEXT.getName())) {
+            logAndGetErrorCallback("JSON response is missing X-Druid-Response-Context", query);
+            return;
+        }
+        JsonNode druidResponseContext = json.get(DruidJsonResponseContentKeys.DRUID_RESPONSE_CONTEXT.getName());
+        if (!druidResponseContext.has(DruidJsonResponseContentKeys.UNCOVERED_INTERVALS.getName())) {
+            logAndGetErrorCallback(
+                    "JSON response is missing 'uncoveredIntervals' from X-Druid-Response-Context header",
+                    query
+            );
+            return;
+        }
+        if (!druidResponseContext.has(DruidJsonResponseContentKeys.UNCOVERED_INTERVALS_OVERFLOWED.getName())) {
+            logAndGetErrorCallback(
+                    "JSON response is missing 'uncoveredIntervalsOverflowed' from X-Druid-Response-Context header",
+                    query
+            );
+            return;
+        }
+        if (!json.has(DruidJsonResponseContentKeys.STATUS_CODE.getName())) {
+            logAndGetErrorCallback("JSON response is missing response status code", query);
+        }
```

Comment:
All of these error strings should be moved into constants and referenced by the constants in the tests so that they'll stay in sync.